src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Count.cs

```diff
@@ -29,6 +29,24 @@ public int Count(string input)
             return count;
         }
 
+        /// <summary>
```

Comment:
We could consider whether there's meaningful benefit to keeping this and the above overloads separate, or whether we should just change the above method to null check the input and then delegate to this one.  I think the only benefit to keeping them separate is if we really wanted to support Count(string) being used with .NET Framework precompiled regexes.

Reply:
I think if we decide to do this then we should probably do it in a separate PR. While I'm not opposed to doing so, I think it would be best to follow the precedence we have been establishing of supporting string overloads on CompileToAssembly case (even for new API) when possible and just throw for the span-based ones.

Reply:
ok